pytensor/tensor/math.py

```diff
@@ -1571,6 +1574,38 @@ def std(input, axis=None, ddof=0, keepdims=False, corrected=False):
     return ret
 
 
+def median(input, axis=None):
+    """
+    Computes the median along the given axis(es) of a tensor `input`.
+
+    Parameters
+    ----------
+    axis: None or int or (list of int) (see `Sum`)
+        Compute the median along this axis of the tensor.
+        None means all axes (like numpy).
+
+    Notes
+    -----
+    This function uses the numpy implementation of median.
+    """
+
+    input = as_tensor_variable(input)
+    sorted_input = input.sort(axis=axis)
+    shape = input.shape[axis]
+    k = extract_constant(shape) // 2
+    if extract_constant(shape % 2) == 0:
```

Comment:
We don't need to force a constant, can use `ifelse`

Reply:
you mean we can use ```ifelse``` op to check whether the shape object is even or not?

Reply:
You can use `ifelse` to emulate a lazy/symbolic if statement like you have here. If the shape is constant, it will be removed, but if not, it will do the right thing at runtime